static/app/views/explore/components/schemaHintsList.tsx

```diff
@@ -142,6 +144,16 @@ function SchemaHintsList({
             ),
             {
               ariaLabel: t('Schema Hints Drawer'),
+              shouldCloseOnLocationChange: newLocation => {
+                return (
+                  location.pathname !== newLocation.pathname ||
+                  // will close if anything but the filter query has changed
+                  !isEqual(
+                    {...location.query, query: {}},
+                    {...newLocation.query, query: {}}
```

Comment:
nit: just a bit cleaner
```suggestion
                    omit(location.query, ['query']),
                    omit(newLocation.query, ['query'])
```